bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -3401,4 +3396,38 @@ public boolean contains(Object parent, Object element) {
 		return false;
 	}
 
+	/**
+	 * Transactionally perform an action on the TreeViewer without redrawing
+	 * regularly. Disables redraw, executes toExecute and enables redraw again.
+	 *
+	 * @param toExecute the transaction to execute on the TreeViewer while it is not
+	 *                  reDrawing
+	 *
+	 */
+	private void executeWithRedrawDisabled(Runnable toExecute) {
+		executeWithRedrawDisabled(toExecute, true);
+	}
+
+	/**
+	 * Transactionally perform an action on the TreeViewer without redrawing
+	 * regularly. Disables redraw, executes toExecute and enables redraw again.
+	 *
+	 * @param toExecute the transaction to execute on the TreeViewer while it is not
+	 *                  reDrawing
+	 *
+	 */
+	private void executeWithRedrawDisabled(Runnable toExecute, boolean shouldDisableRedraw) {
+		Control control = getControl();
+		try {
+			if (shouldDisableRedraw) {
+				control.setRedraw(false);
+			}
+			toExecute.run();
+		} finally {
+			if (shouldDisableRedraw) {
+				control.setRedraw(true);
+			}
+		}
+	}
```

Comment:
This method is currently misleading, as its name states that it disables redraw while it accepts an argument that defines whether it actually does so.
I propose either of the folowing:
- Inline this code as it is only used once.
- Rename the method to something like `execute(Runnable toExecute, boolean withRedrawDisabled)`. In that case, the Javadoc has to be corrected as well (the second parameter is not documented).